### PR TITLE
V2 : enhance SVG deserializer for mirror transforms

### LINKED ISSUE #523: SVG : implement 'mirror' transform
The SVG specification allows negative values for scale() transformation. This is equvilant to the 'mirror' transform, almost.

## Expected Behavior

See 'Scale as a Mirror Function' here: http://tutorials.jenkov.com/svg/svg-transformation.html#scale

## Actual Behavior

An error is thrown due to negative values for scale()

## Steps to Reproduce the Problem

  1. Import SVG file with negative values for scale()

## Specifications

  - Version: ALL
  - Platform: ALL
  - Environment: ALL

### LINKED ISSUE #704: V2 : SVG transformations are incorrect
## Expected Behavior

Translate the attached SVG file. It should look like the attached image in JSCAD.

## Actual Behavior

Pretty messed up.

## Steps to Reproduce the Problem

  1. Drag and drop the attached SVG file to JSCAD.

## Specifications

  - Version: V2
  - Platform: All
  - Environment: All

[fabric_rects.svg.txt](https://github.com/jscad/OpenJSCAD.org/files/5316305/fabric_rects.svg.txt)

### PR DESCRIPTION
**Pull request by z3dev**
These changes enhance the SVG deserializer to support mirror transforms. 

Some small changes were also made:
- corrected order of transforms
- corrected translate transforms with a single value
- added conversion of X / Y attributes on GROUP / USE entities into translate transforms 
- corrected size of rectangles when instantiating to objects

Fixes #704 
Fixes #523 

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [X] Does your submission pass tests?

Additional testing with the SVG sample files was also conducted via the WEB UI.

### CODE REVIEW

File: packages/io/svg-deserializer/index.js [hunk 1]
```diff
@@ -161,23 +161,40 @@ const objectify = (options, group) => {
     shapes = shapes.map((shape) => {
       if ('transforms' in obj) {
         // NOTE: SVG specifications require that transforms are applied in the order given.
+        // But these are applied in the order as required by JSCAD
+        let tr = null
```
Comment (SimonClark) [inline packages/io/svg-deserializer/index.js:165]: Would be nice to see these a bit more readably named as 'transformRotation' or something similar

**Review by SimonClark — APPROVED** [review]
Sorry for the slow response. Life. 1 minor readability nitpick. Review is approved and I'l let you decide to change or merge.

**Comment by z3dev** [comment]
@SimonClark thanks again for the review. i don't mind the nitpicks as readable/maintainable code is best.

code changes made. back to you. please merge if satisfied. 👍

**Review by SimonClark — APPROVED** [review]
Looks good!

**Pull request merged**